src/client/datascience/interactive-common/notebookServerProvider.ts

```diff
@@ -56,23 +68,27 @@ export class NotebookServerProvider implements IJupyterServerProvider {
             this.ui.disableUI = false;
         }
         options.ui.onDidChangeDisableUI(() => (this.ui.disableUI = options.ui.disableUI), this, this.disposables);
-
-        if (!this.serverPromise) {
+        const property = options.localOnly ? 'local' : 'remote';
+        if (!this.serverPromise[property]) {
```

Comment:
Caching local & remote seprately (this is so that you can start jupyter locally - non-raw as well as connect to remote jupyter).